diff --git a/Source/GSEasyHandle.m b/Source/GSEasyHandle.m
index 576a41205..7eb15d512 100644
--- a/Source/GSEasyHandle.m
+++ b/Source/GSEasyHandle.m
@@ -173,6 +173,24 @@ @implementation GSEasyHandle
   struct curl_slist          *_headerList;
 }
 
+#ifdef __ANDROID__
+static NSData *CABundleData() {
+  static NSData *result = nil;
+  static dispatch_once_t predicate;
+
+  dispatch_once(&predicate, ^{
+    NSString *caBundlePath = [[NSBundle mainBundle] pathForResource:@"cacert" ofType:@"pem"];
+    if (caBundlePath) {
+      result = [[NSData alloc] initWithContentsOfFile:caBundlePath];
+    } else {
+      NSLog(@"Warning: missing CA bundle path (cacert.pem) in app bundle");
+    }
+  });
+
+  return result;
+}
+#endif
+
 - (instancetype) initWithDelegate: (id<GSEasyHandleDelegate>)delegate 
 {
   if (nil != (self = [super init])) 
@@ -184,6 +202,18 @@ - (instancetype) initWithDelegate: (id<GSEasyHandleDelegate>)delegate
       _errorBuffer = memset(eb, 0, sizeof(char) * (CURL_ERROR_SIZE + 1));
       
       [self setupCallbacks];
+
+#ifdef __ANDROID__
+      // set CA certificate store (must be stored in app bundle)
+      NSData *caBundleData = CABundleData();
+      if (caBundleData) {
+        struct curl_blob blob;
+        blob.data = (void *)caBundleData.bytes;
+        blob.len = caBundleData.length;
+        blob.flags = CURL_BLOB_COPY;
+        handleEasyCode(curl_easy_setopt(_rawHandle, CURLOPT_CAINFO_BLOB, &blob));
+      }
+#endif
     }
 
   return self;
