diff --git a/Source/NSRunLoop.m b/Source/NSRunLoop.m
index 1746fa3d0..76f2ba7af 100644
--- a/Source/NSRunLoop.m
+++ b/Source/NSRunLoop.m
@@ -749,6 +749,7 @@ + (NSRunLoop*) _runLoopForThread: (NSThread*) aThread
       if (nil != current && [GSCurrentThread() isMainThread] == YES)
         {
           NSAutoreleasePool		*arp = [NSAutoreleasePool new];
+#if 0 // disabled housekeeping timer until we have libffi on Android
           NSNotificationCenter	        *ctr;
           NSNotification		*not;
           NSInvocation		        *inv;
@@ -774,6 +775,7 @@ + (NSRunLoop*) _runLoopForThread: (NSThread*) aThread
                                            userInfo: nil
                                             repeats: YES];
           [current addTimer: timer forMode: NSDefaultRunLoopMode];
+#endif
 
           #ifdef RL_INTEGRATE_DISPATCH
           // We leak the queue drainer, because it's integral part of RL
