diff --git a/src/semaphore.c b/src/semaphore.c
index 5fea9426..738a3a50 100644
--- a/src/semaphore.c
+++ b/src/semaphore.c
@@ -87,7 +87,7 @@ _dispatch_semaphore_debug(dispatch_object_t dou, char *buf, size_t bufsiz)
 			dsema->dsema_sema);
 #endif
 	offset += dsnprintf(&buf[offset], bufsiz - offset,
-			"value = %" PRId64 ", orig = %" PRId64 " }", dsema->dsema_value, dsema->dsema_orig);
+			"value = %" PRIdPTR ", orig = %" PRIdPTR " }", dsema->dsema_value, dsema->dsema_orig);
 	return offset;
 }
 
@@ -272,7 +272,7 @@ _dispatch_group_debug(dispatch_object_t dou, char *buf, size_t bufsiz)
 			dg->dg_sema);
 #endif
 	offset += dsnprintf(&buf[offset], bufsiz - offset,
-			"count = %" PRId64 ", waiters = %d }", dg->dg_value, dg->dg_waiters);
+			"count = %" PRIdPTR ", waiters = %d }", dg->dg_value, dg->dg_waiters);
 	return offset;
 }
 
