From 0c260845173a2838a4d6ead5849251728afa9843 Mon Sep 17 00:00:00 2001
From: hmelder <service@hugomelder.com>
Date: Thu, 1 May 2025 14:14:55 +0800
Subject: [PATCH] Bump robin map version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9aa16b5..7be34400 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,7 +118,7 @@ if (NOT tsl-robin-map_FOUND)
 	FetchContent_Declare(
 		robinmap
 		GIT_REPOSITORY https://github.com/Tessil/robin-map/
-		GIT_TAG        v1.2.1)
+		GIT_TAG        v1.4.0)
 
 	FetchContent_MakeAvailable(robinmap)
 endif()
