https://github.com/gnustep/libobjc2/pull/231

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 559555d..8ac86a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,7 +228,7 @@ function(test_cxx CXX_RUNTIME_NAME IS_STDLIB)
 		message(STATUS "Testing ${CXX_RUNTIME_LIB} as the C++ runtime library")
 		if (ANDROID)
 			# pass on Android toolchain flags
-			set(CXX_RUNTIME_FLAGS "-DANDROID_NDK=${ANDROID_NDK}" "-DANDROID_ABI=${ANDROID_ABI}" "-DANDROID_ARM_MODE=${ANDROID_ARM_MODE}" "-DANDROID_LD=${ANDROID_LD}" "-DANDROID_PLATFORM=${ANDROID_PLATFORM}" "-DANDROID_STL=${ANDROID_STL}")
+			set(CXX_RUNTIME_FLAGS "-DANDROID_NDK=${ANDROID_NDK}" "-DANDROID_ABI=${ANDROID_ABI}" "-DANDROID_ARM_MODE=${ANDROID_ARM_MODE}" "-DANDROID_ARM_NEON=${ANDROID_ARM_NEON}" "-DANDROID_LD=${ANDROID_LD}" "-DANDROID_PLATFORM=${ANDROID_PLATFORM}" "-DANDROID_STL=${ANDROID_STL}")
 		endif()
 		try_compile(USERUNTIME 
 			"${CMAKE_BINARY_DIR}/CMake"
